plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt

```diff
@@ -777,9 +790,13 @@ class JKCodeBuilder(context: NewJ2kConverterContext) {
 
         override fun visitKtPrimaryConstructorRaw(ktPrimaryConstructor: JKKtPrimaryConstructor) {
             ktPrimaryConstructor.annotationList.accept(this)
-            renderModifiersList(ktPrimaryConstructor)
+            val redundantConstructor = redundantModifiersForConstructor(ktPrimaryConstructor)
+            if (!redundantConstructor) {
+                renderModifiersList(ktPrimaryConstructor)
```

Comment:
`redundantModifiersForConstructor` checks only visibility modifiers, but here we skip rendering any modifiers. Is this intended?